saleor/shipping/models.py

```diff
@@ -96,8 +96,14 @@ def applicable_shipping_methods_by_channel(self, shipping_methods, channel_id):
             "price_amount"
         )
 
+    def exclude_shipping_methods_for_excluded_products(
+        self, qs, product_ids: List[int]
+    ):
+        """Exclude the ShippingMethods which have excluded given products."""
+        return qs.exclude(excluded_products__id__in=product_ids)
+
     def applicable_shipping_methods(
-        self, price: Money, channel_id, weight, country_code
+        self, price: Money, channel_id, weight, country_code, excluded_product_ids=None
```

Comment:
Let's change `excluded_product_ids` to `product_ids` and add a comment in code to explain what it does.